### PR TITLE
The scope of the unsafe block can be appropriately reduced

### PR DESCRIPTION
**Pull request by peamaeq**

### CODE REVIEW

File: tokio/src/io/util/vec_with_initialized.rs [hunk 1]
```diff
@@ -107,8 +107,8 @@ where
         // Note that this relies on the fact that `V` is either `Vec<u8>` or
         // `&mut Vec<u8>`, so the vector returned by `self.vec.as_mut()` cannot
         // change from call to call.
+        self.num_initialized = parts.initialized;
         unsafe {
-            self.num_initialized = parts.initialized;
             vec.set_len(parts.len);
         }
```
Comment (Darksonn) [inline tokio/src/io/util/vec_with_initialized.rs:113]: I'm not so sure that this makes the code easier to read. The assignment to `num_initialized` is important for the safety of the `set_len` call too.

File: tokio/src/process/windows.rs [hunk 2]
```diff
@@ -152,13 +152,13 @@ impl AsRawHandle for Child {
...
+            let rc = unsafe { UnregisterWaitEx(self.wait_object, INVALID_HANDLE_VALUE) };
             if rc == 0 {
                 panic!("failed to unregister: {}", io::Error::last_os_error());
             }
-            drop(Box::from_raw(self.tx));
-        }
+            drop(unsafe { Box::from_raw(self.tx) });
+        
     }
 }
```
Comment (Darksonn) [inline tokio/src/process/windows.rs:163]: This needs some rustfmt.

**Comment by Darksonn** [comment]
There are some more things that need rustfmt. You can fix them with the following command:
```
# Mac or Linux
rustfmt --edition 2018 $(git ls-files '*.rs')

# Powershell
Get-ChildItem . -Filter "*.rs" -Recurse | foreach { rustfmt --edition 2018 $_.FullName }
```

File: tokio/src/io/util/vec_with_initialized.rs (hunk 1 above)
Comment (Finomnis) [reply tokio/src/io/util/vec_with_initialized.rs:113]: I partially disagree, as checking for soundness includes more than just looking at the unsafe blocks. If there's an `unsafe` around the entire function code, it's easy to deduct that everything in the function has to be considered for soundness.

**Comment by Darksonn** [comment]
You will need to merge in master to fix the CI failures.

**Review by Darksonn — APPROVED** [review]

**Comment by Darksonn** [comment]
I'm not really sure why CI failed here.

**Pull request closed without merging**